File: frontend/src/types/timetable.ts
export interface SlotInfo {
  subject: string; teacher: string | null; room: string | null; is_lab: boolean;
}
export type ClassTimetable = Record<string, Record<string, Record<string, SlotInfo>>>
export interface TimetableResponse {
  job_id: string; timetable: ClassTimetable;
  teacher_timetable: Record<string, Record<string, Record<string, { class: string; subject: string; room: string }>>>;
  room_timetable: Record<string, Record<string, Record<string, { class: string; subject: string; teacher: string }>>>;
}

File: backend/api/schemas/timetable.py
"""Pydantic schemas for Timetable grid responses."""

from typing import Optional

from pydantic import BaseModel


class SlotInfo(BaseModel):
    subject: str
    teacher: Optional[str] = None
    room: Optional[str] = None
    is_lab: bool = False


class ValidationResponse(BaseModel):
    valid: bool
    total_slots: Optional[int] = None
    sections: Optional[int] = None
    teachers: Optional[int] = None
    rooms: Optional[int] = None
    errors: list[str] = []
    warnings: list[str] = []


class ImportSummary(BaseModel):
    institution_id: str
    inserted: dict
